pkg/controller/operators/catalog/operator.go

```diff
@@ -416,6 +505,27 @@ func NewOperator(ctx context.Context, kubeconfigPath string, clock utilclock.Clo
 	jobInformer := k8sInformerFactory.Batch().V1().Jobs()
 	sharedIndexInformers = append(sharedIndexInformers, jobInformer.Informer())
 
+	if err := labelObjects(batchv1.SchemeGroupVersion.WithResource("jobs"), jobInformer.Informer(), labeller.ObjectLabeler[*batchv1.Job, *batchv1applyconfigurations.JobApplyConfiguration](
+		ctx, op.logger, func(object metav1.Object) bool {
+			job, ok := object.(*batchv1.Job)
+			if !ok {
+				return false
```

Comment:
Would we ever get the wrong type here?

Reply:
No, but I'd rather write code that did not panic and crash-loop just in case. What would you suggest?